modules/idlgen/core/src/main/scala/avro/AvroIdlGenerator.scala

```diff
@@ -92,19 +104,24 @@ trait AvroIdlGenerator extends IdlGenerator {
     }
   }
 
-  private def mappedType(typeArg: Type.Arg): AvroType = typeArg match {
-    case targ"Boolean"    => "boolean"
-    case targ"Int"        => "int"
-    case targ"Long"       => "long"
-    case targ"Float"      => "float"
-    case targ"Double"     => "double"
-    case targ"String"     => "string"
-    case targ"Seq[$t]"    => AvroArray(mappedType(t))
-    case targ"List[$t]"   => AvroArray(mappedType(t))
-    case targ"Array[$t]"  => AvroArray(mappedType(t))
-    case targ"Option[$t]" => AvroOption(mappedType(t))
-    case targ"Empty.type" => AvroEmpty
-    case _                => typeArg.toString
+  def mappedType(typeArg: Tree): AvroType = typeArg match {
+    case ast._Ident(Ident(TypeName("Boolean"))) => "boolean"
```

Comment:
For this six cases, in which you have very similar patterns, I would use an [extractor object](https://docs.scala-lang.org/tour/extractor-objects.html), called `BasicTree`, so you may turn the first case into 
```Scala
case BaseType("Int") => "int"
```
This is even better in the five applied-tree cases, in which you could write.
```Scala
case SingleAppliedTree( "Seq", t) => AvroArray(mappedType(t))
```

Reply:
Thanks! I was definitely not happy with those humongous matches, but didn't thought of an extractor :)